File: app/src/main/java/com/example/android/tj/model/TJServiceSearchResult.java
package com.example.android.tj.model;

import android.support.annotation.NonNull;
import android.util.Pair;

import com.google.gson.Gson;

import java.util.List;
import java.util.stream.Collectors;

public class TJServiceSearchResult {
    public List<String> fileNames;
    public List<String> hashes;


    public TJServiceSearchResult(List<Pair<String, String>> pairs) {
        this.hashes = pairs.stream().map(pair -> pair.first).collect(Collectors.toList());
        this.fileNames = pairs.stream().map(pair -> pair.second).collect(Collectors.toList());
    }

    @NonNull
    public String toString() {
        return new Gson().toJson(this);
    }

    public static TJServiceStatus fromJson(String jsonStr) {
        return new Gson().fromJson(jsonStr, TJServiceStatus.class);
    }
}

File: app/src/main/java/com/example/android/tj/model/TJServiceSearchResult.kt
package com.example.android.tj.model

import android.util.Pair
import com.google.gson.Gson

@Deprecated("to be removed")
class TJServiceSearchResult(pairs: List<Pair<String, String>>) {
    var fileNames: List<String>
    var hashes: List<String>


    init {
        this.hashes = pairs.map { it.first }
        this.fileNames = pairs.map { it.second }
    }

    override fun toString(): String {
        return Gson().toJson(this)
    }

    companion object {

        fun fromJson(jsonStr: String): TJServiceStatus {
            return Gson().fromJson(jsonStr, TJServiceStatus::class.java)
        }
    }
}
